thunder/executors/torch_compile.py

```diff
@@ -174,6 +168,10 @@ def _can_fuse_node(n: Node):
 
         fusedtrace.bound_symbols = fused_bsyms
 
+        fusedtrace = rematerialize(fusedtrace)
+        fusedtrace = dce(fusedtrace)
+        fusedtrace = update_fusion_call_ctx(fusedtrace)
```

Comment:
I added this because nvfuser does it, but if reviewers think it's a bad idea let's drop it.